test/smart-contract-tests/updateContract.test.ts

```diff
@@ -34,7 +36,11 @@ test('update contract with the wrong private key', async () => {
     };
 
     const receiveName = 'INDBank.insertAmount';
-    const params = Buffer.from([]);
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
```

Comment:
@shjortConcordium so for any test case we need to provide schema as input? also how do we get the schema for a specific smart contract using node-sdk or do we need to use the schema generated using the concordium-client?

Reply:
For a contract with the schema embedded, you would use getModuleSource, where the schema would be contained in the returned bytes, but that needs to be decoded.
concordium-client uses this, in the `module inspect` and `contract show` commands, where it can show the schema.

Otherwise you need the schema file, which can be generated when you build your smart contract, using the "-schema-out" flag.

I think it would make sense to mainly support providing the schema file as input. Getting an embedded one can be done separately.

About schemas:
https://developer.concordium.software/en/mainnet/smart-contracts/general/contract-schema.html#contract-schema 
Building schemas:
https://developer.concordium.software/en/mainnet/smart-contracts/guides/build-schema.html

Reply:
Now we have finished all types except the enum type. We have created test cases for all basic types like u8, u16, u32, u64, u128, i8, i16, i32, i64, i128, array and string.